cloudigrade/account/util.py

```diff
@@ -45,3 +49,111 @@ def create_initial_aws_instance_events(account, instances_data):
             event.save()
             saved_instances[region].append(instance)
     return dict(saved_instances)
+
+
+def create_new_machine_images(account, instances_data):
+    """
+    Create AwsMachineImage that have not been seen before.
+
+    Args:
+        account (AwsAccount): The account associated with the machine image
+        instances_data (dict): Dict whose keys are AWS region IDs and values
+            are each a list of dictionaries that represent an instance
+
+    Returns:
+        list: A list of image ids that were added to the database
+
+    """
+    saved_amis = []
+    for __, instances in instances_data.items():
+        platforms = {instance['ImageId']: instance['Platform']
+                     for instance in instances
+                     if instance.get('Platform') == 'Windows'}
+        seen_amis = set([instance['ImageId'] for instance in instances])
+        known_amis = AwsMachineImage.objects.filter(
+            ec2_ami_id__in=list(seen_amis)
+        )
+        new_amis = list(seen_amis.difference(known_amis))
+
+        for new_ami in new_amis:
+            ami = AwsMachineImage(
+                account=account,
+                ec2_ami_id=new_ami,
+                platform=(MachineImage.TYPE.Windows
+                          if platforms.get(new_ami)
+                          else MachineImage.TYPE.Linux)
```

Comment:
I feel lukewarm about explicitly setting "Linux" when we only really know if it's Windows or not-Windows at this point.

Maybe we need an "Unknown" state or to let it be nullable until we know more information.

Admittedly, it's *very* unlikely to be neither Windows or Linux, but the possibility exists.

Reply:
That's actually why I ended up making the model use a choice with Unix-like (what if they're running FreeBSD). But it's really about distinguishing windows or not windows. Would you prefer a boolean flag there instead of a choice?

Reply:
The more I think about it, the more I think a simple is-Windows boolean makes sense. That would cover exactly the case we currently need to know about (i.e. skipping AMIs for Windows), doesn't introduce ambiguity, and is easy to migrate/convert to a richer "choice" field later if we see value in it.